packages/back-end/src/models/DataSourceModel.ts

```diff
@@ -61,9 +61,15 @@ export async function getInstallationDatasources(): Promise<
     throw new Error("Cannot get all installation data sources in cloud mode");
   }
   if (usingFileConfig()) {
-    const organizationId = (await findAllOrganizations(0, "", 1))
-      .organizations[0].id;
-    return getConfigDatasources(organizationId);
+    // For multi-org self-hosted using config file, we don't care about which org the
+    // datasource is assigned to, so getting any arbitrary one is fine.  All organizations
+    // share the same datasource config.
+    const organizationId = (await getSelfHostedOrganization())?.id;
+    if (organizationId) {
+      return getConfigDatasources(organizationId);
+    } else {
+      throw new Error("No self-hosted organization found");
+    }
```

Comment:
If you look at the `getConfigDatasources` method, it only uses the org id to populate that field in the returned data sources.  So you can just pass in an empty string and it will be fine.  No need to do a Mongo lookup here and throw, potentially causing even more license error edge cases (e.g. someone sets a license env var before creating their first org).

Also, we should change the `getLicenseMetaData` function so it never throws (wrap the contents in try/catch).  We never want to block getting a license key just because there's an error gathering metadata.

Reply:
Good points.